pulp_ansible/app/downloaders.py

```diff
@@ -91,12 +101,15 @@ async def _run(self, extra_data=None):
             # Galaxy Token
             headers = {"Authorization": self.ansible_token}
         else:
-            token = await self.update_token_from_auth_url()
+            token = self.registry_auth["bearer"]
             # Keycloak Token
             headers = {"Authorization": "Bearer {token}".format(token=token)}
 
         async with self.session.get(self.url, headers=headers, proxy=self.proxy) as response:
-            self.raise_for_status(response)
+            self.raise_for_status(response, handle_401)
+            if response.status == 401:
+                await self.update_token_from_auth_url()
+                return await self._run(handle_401=False)
```

Comment:
This could run a little unexpectedly in cases where a lot of other downloaders are running L#111 waits a long time, and then this one is the first one to wake up and it won't re-acquire the token even though it should.